pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -448,6 +449,13 @@ public List<InstanceConfig> getAllHelixInstanceConfigs() {
     return HelixHelper.getInstanceConfigs(_helixZkManager);
   }
 
+  /**
+   * returns server to segments count
+   */
+  public Map<String, Integer> getServerToSegmentCountMap() {
+    return _serverToSegmentCountMap;
```

Comment:
I don’t think this is a right way to achieve this as it depends on the invocation of `getServerToSegmentsMap`.
You can have the same code logic as `getServerToSegmentsMap` and aggregate only the count.
`getServerToSegmentsMap` can call this when `detailed` is false and compute the count from the response of `getServerToSegmentsMap` itself  when `detailed` is true

Reply:
I tried to calculate _serverToSegmentCountMap through getServerToSegmentsMap itself which I called both for detailed = true and detailed = false. Added details of the count only after calculating it in the api.

Reply:
This is not thread safe. We cannot cache this map here. In verbose mode, we should get the `serverToSegmentsMap` and then use it to calculate the `serverToSegmentCountMap` in the restlet resource